services/filesstore/s3store.go

```diff
@@ -238,9 +238,13 @@ func (b *S3FileBackend) ListDirectory(path string) (*[]string, *model.AppError)
 	}
 
 	doneCh := make(chan struct{})
-
 	defer close(doneCh)
 
+	if !strings.HasSuffix(path, "/") && len(path) > 0 {
```

Comment:
Are there s3store unit tests we can improve here?

Reply:
i already updated the filesstore unit test that tests this case
https://github.com/mattermost/mattermost-server/pull/11657/files#diff-f86aca3f016fa314f6ce7b800efdb74aR257